File: import-molecules.py
#!/usr/bin/env python3
import csv
import mysql.connector
import os
from datetime import datetime

# Connexion à la base de données
db = mysql.connector.connect(
    host=os.getenv('DB_HOST', 'gateway01.us-east-1.prod.aws.tidbcloud.com'),
    port=int(os.getenv('DB_PORT', 4000)),
    user=os.getenv('DB_USER'),
    password=os.getenv('DB_PASSWORD'),
    database=os.getenv('DB_NAME'),
    ssl_ca='/etc/ssl/certs/ca-certificates.crt'
)

cursor = db.cursor()

print("🔄 Import des nouvelles molécules...\n")

# Lire le fichier CSV
with open('/home/ubuntu/perfumum-research/NOUVELLES_MOLECULES_25.csv', 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    count = 0
    
    for row in reader:
        if not row['name']:  # Skip empty rows
            continue
            
        try:
            # Préparer les données
            name = row['name']
            family = row['family'] or None
            formula = row['chemicalFormula'] or None
            olfactive_profile = row['olfactiveProfile'] or None
            emotional_resonance = row['emotionalResonance'] or None
            functional_effect = row['functionalEffect'] or None
            source_origin = row['sourceOrigin'] or None
            
            # Radar values
            radar_intensity = int(row['radarIntensity']) if row['radarIntensity'] else None
            radar_freshness = int(row['radarFreshness']) if row['radarFreshness'] else None
            radar_warmth = int(row['radarWarmth']) if row['radarWarmth'] else None
            radar_sweetness = int(row['radarSweetness']) if row['radarSweetness'] else None
            radar_spiciness = int(row['radarSpiciness']) if row['radarSpiciness'] else None
            radar_earthiness = int(row['radarEarthiness']) if row['radarEarthiness'] else None
            
            # Physical properties
            molecular_weight = int(row['molecularWeight']) if row['molecularWeight'] else None
            boiling_point = int(row['boilingPoint']) if row['boilingPoint'] else None
            volatility = int(row['volatility']) if row['volatility'] else None
            intensity = int(row['intensity']) if row['intensity'] else None
            complexity = int(row['complexity']) if row['complexity'] else None
            
            # Additional info
            botanical_sources = row['botanicalSources'] or None
            extraction_method = row['extractionMethod'] or None
            therapeutic_properties = row['therapeuticProperties'] or None
            
            # Insérer dans la base de données
            query = """
            INSERT INTO molecules (
                name, family, chemicalFormula, olfactiveProfile, emotionalResonance,
                functionalEffect, sourceOrigin, radarIntensity, radarFreshness, radarWarmth,
                radarSweetness, radarSpiciness, radarEarthiness, molecularWeight, boilingPoint,
                volatility, intensity, complexity, botanicalSources, extractionMethod,
                therapeuticProperties, createdAt, updatedAt
            ) VALUES (
                %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, NOW(), NOW()
            )
            """
            
            values = (
                name, family, formula, olfactive_profile, emotional_resonance,
                functional_effect, source_origin, radar_intensity, radar_freshness, radar_warmth,
                radar_sweetness, radar_spiciness, radar_earthiness, molecular_weight, boiling_point,
                volatility, intensity, complexity, botanical_sources, extraction_method,
                therapeutic_properties
            )
            
            cursor.execute(query, values)
            db.commit()
            
            count += 1
            print(f"✅ {name} importée (ID: {cursor.lastrowid})")
            
        except Exception as e:
            print(f"❌ Erreur pour {row.get('name', 'Unknown')}: {str(e)}")
            db.rollback()

print(f"\n✨ Import terminé ! {count} molécules importées.")

cursor.close()
db.close()

File: import-molecules.mjs
import { drizzle } from 'drizzle-orm/mysql2';
import mysql from 'mysql2/promise';
import * as schema from './drizzle/schema.js';
import fs from 'fs';

const connection = await mysql.createConnection(process.env.DATABASE_URL);
const db = drizzle(connection, { schema, mode: 'default' });

// Lire et parser le CSV
const csvContent = fs.readFileSync('/home/ubuntu/perfumum-research/NOUVELLES_MOLECULES_25.csv', 'utf-8');
const lines = csvContent.split('\n');
const headers = lines[0].split(',');

console.log('🔄 Import des nouvelles molécules...\n');

let count = 0;

for (let i = 1; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line) continue;
  
  // Parse CSV line (handle quoted fields)
  const values = [];
  let current = '';
  let inQuotes = false;
  
  for (let char of line) {
    if (char === '"') {
      inQuotes = !inQuotes;
    } else if (char === ',' && !inQuotes) {
      values.push(current);
      current = '';
    } else {
      current += char;
    }
  }
  values.push(current);
  
  if (values.length < 10) continue; // Skip invalid lines
  
  const molecule = {
    name: values[0] || null,
    family: values[1] || null,
    chemicalFormula: values[2] || null,
    olfactiveProfile: values[3] || null,
    emotionalResonance: values[4] || null,
    functionalEffect: values[5] || null,
    sourceOrigin: values[6] || null,
    radarIntensity: values[7] ? parseInt(values[7]) : null,
    radarFreshness: values[8] ? parseInt(values[8]) : null,
    radarWarmth: values[9] ? parseInt(values[9]) : null,
    radarSweetness: values[10] ? parseInt(values[10]) : null,
    radarSpiciness: values[11] ? parseInt(values[11]) : null,
    radarEarthiness: values[12] ? parseInt(values[12]) : null,
    molecularWeight: values[13] ? parseInt(values[13]) : null,
    boilingPoint: values[14] ? parseInt(values[14]) : null,
    volatility: values[15] ? parseInt(values[15]) : null,
    intensity: values[16] ? parseInt(values[16]) : null,
    complexity: values[17] ? parseInt(values[17]) : null,
    botanicalSources: values[18] || null,
    extractionMethod: values[19] || null,
    therapeuticProperties: values[20] || null
  };
  
  if (!molecule.name) continue;
  
  try {
    const [result] = await db.insert(schema.molecules).values(molecule);
    count++;
    console.log(`✅ ${molecule.name} importée (ID: ${result.insertId})`);
  } catch (error) {
    console.error(`❌ Erreur pour ${molecule.name}:`, error.message);
  }
}

console.log(`\n✨ Import terminé ! ${count} molécules importées.`);
await connection.end();
process.exit(0);
